src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/acl_editor.cljs

```diff
@@ -24,78 +24,114 @@
-            :style {:float "right" :width 200 :height 33 :marginLeft "1em"}
+            :style {:display "inline-block" :width 200 :height 33 :marginLeft "1rem" :marginBottom 0}
+            :disabled (= (:email acl-entry) (:userEmail @utils/current-user-info))
             :value (:accessLevel acl-entry)
             :onChange #(swap! state assoc-in [:non-project-owner-acl-vec i :accessLevel]
                               (.. % -target -value))}
            access-levels)
-          (common/clear-both)])
+          (when (:pending? acl-entry)
+            [:span {:style {:fontStyle "italic" :color (:text-light style/colors)
+                            :marginLeft "0.5rem" }}
+             "Pending..."])])
        (:non-project-owner-acl-vec @state))
-      [:div {:style {:marginBottom "0.5em"}}
+      [:div {:style {:margin "0.5rem 0"}}
        [comps/Button {:text "Add new" :icon :add
                       :onClick #(swap! state update-in [:non-project-owner-acl-vec]
                                        conj {:email "" :accessLevel "READER"})}]]
       (style/create-validation-error-message (:validation-error @state))
       [comps/ErrorViewer {:error (:save-error @state)}]])
     :ok-button {:text "Save" :onClick #(react/call :persist-acl component)}}])
 
+(defn- render-invite-offer [props state component]
+  [modal/OKCancelForm
+   {:header (let [workspace-id (:workspace-id props)]
+              (str "Invite new users to" (:namespace workspace-id) "/" (:name workspace-id)))
+    :content (react/create-element
+              [:div {}
+               (when (:saving? @state)
+                 [comps/Blocker {:banner "Updating..."}])
+               [:div {:style {:padding "0.5rem 0" :fontSize "90%" :marginTop "0.5rem"}}
+                [:div {:style {:display "inline-block" :width 400}} "User ID"]
+                [:div {:style {:display "inline-block" :width 200 :marginLeft "1rem"}} "Access Level"]]
+               (map-indexed
+                (fn [i acl-entry]
+                  [:div {:style {:borderTop style/standard-line :padding "0.5rem 0"}}
+                   [:div {:style {:display "inline-block" :width 400 :fontSize "88%"}}
+                    (:email acl-entry)]
+                   [:span {:style {:display "inline-block" :lineHeight "33px"
+                                   :width 200 :height 33 :verticalAlign "middle"
+                                   :marginLeft "1rem" :marginBottom 0}}
+                    (:accessLevel acl-entry)]])
+                (:users-not-found @state))])
+    :ok-button {:text "Invite" :onClick #(react/call :persist-acl component true)}}])
+
 (react/defc AclEditor
   {:render
    (fn [{:keys [props state this]}]
      (if (or (:non-project-owner-acl-vec @state) (:project-owner-acl-vec @state))
-       (render-acl-content props state this)
+       (if (:offering-invites? @state)
+         (render-invite-offer props state this)
+         (render-acl-content props state this))
```

Comment:
Rather than passing the entire guts of `AclEditor` to these functions, pass just what you need.  Use a callback for the onClicks.